tests/console/pam.pm

```diff
@@ -0,0 +1,66 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+# Summary: This PM is to run a suite of tests about PAM, and gets a TAP
+#          format result. Four files(pam.sh, pam_test, pam_test.sh,
+#          system-default) are included. pam.sh is a "bats" script, and
+#          includes 25 tests for PAM. The other files are from the project
+#          "pam_test" which can be used to test a PAM stack for authentication
+#          and password change. The link: https://github.com/pbrezina/pam-test
+#   Steps:
+#       - add qa-head repo and install the tool "bats" from qa-head repo.
+#         Bats is a TAP-compliant testing framework for Bash.
+#       - download "pam_test" dir(automated tests of PAM) from "data" dir
+#         on openqa host if "prepare_test_data" perl module desn't run.
+#       - run the suite of tests in SUT.
+# Maintainer: Jun Wang <jgwang@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use repo_tools 'add_qa_head_repo';
+
+sub run {
+    select_console 'root-console';
+
+    # bats is from qa repo, so adds qa repo and install bats
+    if (get_var('QA_HEAD_REPO')) {
+        add_qa_head_repo();
+    } else {
+        my $version      = get_required_var('VERSION');
+        my $qa_head_repo = "http://download.suse.de/ibs/QA:/Head/SLE-$version/";
```

Comment:
No, these tests are not tested in openSUSE, and I plan to only run these tests in SLE.
The line that you suggested is directly added into "pam.pm" ?

Reply:
yeah, before L34. Although don't forget about factory first :) (but maybe a ticket to enable them in openSUSE can be added for now)

Reply:
> yeah, before L34. Although don't forget about factory first :) (but maybe a ticket to enable them in openSUSE can be added for now)

It's a good suggestion, I will have a try to open a ticket to enable them in openSUSE.

Reply:
@foursixnine I plan to support openSUSE directly in this PR, and I am testing these tests on openSUSE, so I don't think I need to open a ticket. But I have a question, you know I need the package "pam-test", but there is not this package from "QA_HEAD_REPO"(like this https://download.opensuse.org/repositories/devel:/openSUSE:/QA:/Tumbleweed/openSUSE_Tumbleweed/) in openSUSE jobs, could I take these repos (http://download.suse.de/ibs/QA:/Head/*) instead of these repos (https://download.opensuse.org/repositories/devel:/openSUSE:/QA:/)?

Reply:
I guess the package will need to be linked in the openSUSE:QA repos. @plusky am I right?

Reply:
> I guess the package will need to be linked in the openSUSE:QA repos. @plusky am I right?

I ran the tests on openSUSE, they worked well. Once openSUSE:QA repos are ready, I will update code.  @plusky if openSUSE:QA are ready, please ping me, thank you for helping me a lot :smile: